src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,82 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React from 'react';
+import { Filter, TodoStates } from '../../types/Filter';
+
+function capitalizer(str: string): string {
+  return str.at(0)?.toUpperCase() + str.slice(1);
+}
+
+const todoStatesArray = Object.values(TodoStates);
+
+type Props = {
+  filter: Filter;
+  onFilter: (f: Filter) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ filter, onFilter }) => {
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onFilter({
+      ...filter,
+      select: event.target.value as TodoStates,
+    });
+  };
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onFilter({
+      ...filter,
+      input: event.target.value,
+    });
+  };
+
+  const handleClearInput = () => {
+    onFilter({
+      ...filter,
+      input: '',
+    });
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filter.select}
+            onChange={handleChangeSelect}
+          >
+            {todoStatesArray.map(value => (
+              <option value={value} key={value}>
+                {capitalizer(value)}
+              </option>
+            ))}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={filter.input}
+          onChange={handleChangeInput}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {!!filter.input && (
```

Comment:
```suggestion
        {filter.input && (
```
You don't need to convert it to a boolean, not empty string is already truthy value
https://developer.mozilla.org/en-US/docs/Glossary/Truthy